packages/apollo-server-core/src/runQuery.ts

```diff
@@ -114,7 +115,7 @@ function doRunQuery(options: QueryOptions): Promise<GraphQLResponse> {
   logFunction({ action: LogAction.request, step: LogStep.start });
 
   const context = options.context || {};
-  let extensions = [];
+  let extensions = options.extensions !== undefined ? options.extensions : [];
   if (options.tracing) {
     extensions.push(TracingExtension);
```

Comment:
I wonder if the tracing extension should be applied before all other extensions, otherwise it's going to give misleading numbers (the custom extensions may take a long time to execute, but the tracing would never pick up on that as it'd start recording AFTER they've all executed)

Reply:
Great idea. We should also run "didStop" handlers in reverse order for that reason.